python/hsfs/engine/python.py

```diff
@@ -751,9 +752,23 @@ def _write_dataframe_kafka(
         writer = self._get_encoder_func(feature_group._get_encoded_avro_schema())
 
         def acked(err, msg):
-            if err is not None:
+            if err is not None and offline_write_options.get("debug_kafka", False):
                 print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
-
+            else:
+                try:
+                    if progress_bar is not None:
```

Comment:
I think you can skip this try catch block, or are you expecting a specific error?